### PR TITLE
Fix: add pagination for teams api

### LINKED ISSUE #981: Pagination for `env0_teams` data source
**Describe the bug**
My organization has huge amount of teams (1176 at the moment). The `env0_teams` fails with

```
╷
│ Error: Could not get teams: 502 Bad Gateway: {"message": "Internal server error"}
│ 
│   with data.env0_teams.all_teams,
│   on env0_project_policy.tf line 1, in data "env0_teams" "all_teams":
│    1: data "env0_teams" "all_teams" {}
│ 
╵
```

**To Reproduce**

```
data "env0_teams" "all_teams" {}
```

**Expected behavior**

This should work without error.

**Provider Version**
1.22.1

**Screenshots**
N/A

**Additional context**

It looks like https://github.com/env0/terraform-provider-env0/blob/v1.22.1/env0/data_teams.go#L31 calling this code https://github.com/env0/terraform-provider-env0/blob/v1.22.1/client/team.go#L72-L87 and I am not seeing it attempting to do the pagination. Unless it is buried somewhere in the middleware, as I can see https://github.com/env0/terraform-provider-env0/blob/v1.22.1/client/pagination.go but I don't see where it is being injected into the HTTP client, so I am assuming it is not enabled for this call.

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #981 

### Solution

Added pagination for teams() api call.
Minor updates to code to add the pagination.
Updated unit tests.

### CODE REVIEW

File: client/environment.go [hunk 1]
```diff
@@ -233,24 +233,20 @@ type EnvironmentDeployResponse struct {
 	Id string `json:"id"`
 }
 
-func (Environment) getEndpoint() string {
```
Comment (TomerHeber) [inline client/environment.go:236]: not needed... I do it in a different way.

File: client/pagination.go [hunk 2]
```diff
@@ -29,7 +24,7 @@ func (p *Pagination) next(currentPageSize int) bool {
 }
 
 // params - additional params. may be nil.
-func getAll[P Paginated](client *ApiClient, params map[string]string) ([]P, error) {
+func getAll[P any](client *ApiClient, endpoint string, params map[string]string) ([]P, error) {
```
Comment (TomerHeber) [inline client/pagination.go:27]: modified to support all types.

File: client/team.go [hunk 3]
```diff
@@ -84,20 +84,28 @@ func (client *ApiClient) GetTeams(params map[string]string) ([]Team, error) {
...
 func (client *ApiClient) TeamsByName(name string) ([]Team, error) {
-	return client.GetTeams(map[string]string{"name": name})
+	organizationId, err := client.OrganizationId()
+	if err != nil {
+		return nil, err
+	}
+
+	var result []Team
+
+	if err := client.http.Get("/teams/organizations/"+organizationId, map[string]string{
```
Comment (TomerHeber) [inline client/team.go:104]: name param doesn't work with limit and offset... (bug in backend?)

**Review by razbensimon — APPROVED** [review]
NIce 👍🏼

**Pull request merged**